six/modules/c++/cphd/include/cphd/CPHDXMLControl.h

```diff
@@ -32,86 +31,188 @@
-    XMLElem toXML(const std::string& name, const AntennaParameters &ap,
-            XMLElem parent = NULL);
-    XMLElem toXML(const VectorParameters& obj, XMLElem parent = NULL);
-
-    XMLElem areaLineDirectionParametersToXML(const std::string& name,
-            const AreaDirectionParameters& obj,
-            XMLElem parent = NULL);
-    XMLElem areaSampleDirectionParametersToXML(const std::string& name,
-            const AreaDirectionParameters& obj,
-            XMLElem parent = NULL);
-
-    // Read functions
-    void fromXML(const XMLElem dataXML, Data& obj);
-    void fromXML(const XMLElem globalXML, Global& obj);
-    void fromXML(const XMLElem channelXML, Channel& obj);
-    void fromXML(const XMLElem srpXML, SRP& obj);
-    void fromXML(const XMLElem antennaXML, Antenna& obj);
-    void fromXML(const XMLElem antennaParamsXML, AntennaParameters& params);
-    void fromXML(const XMLElem vectorParametersXML, VectorParameters& obj);
+    //! Write to XML object
+    XMLElem toXML(const CollectionID& obj, XMLElem parent);
+    XMLElem toXML(const Global& obj, XMLElem parent);
+    XMLElem toXML(const SceneCoordinates& obj, XMLElem parent);
+    XMLElem toXML(const Data& obj, XMLElem parent);
+    XMLElem toXML(const Channel& obj, XMLElem parent);
+    XMLElem toXML(const Pvp& obj, XMLElem parent);
+    XMLElem toXML(const SupportArray& obj, XMLElem parent);
+    XMLElem toXML(const Dwell& obj, XMLElem parent);
+    XMLElem toXML(const ReferenceGeometry& obj, XMLElem parent);
+    XMLElem toXML(const Antenna& obj, XMLElem parent);
+    XMLElem toXML(const TxRcv& obj, XMLElem parent);
+    XMLElem toXML(const ErrorParameters& obj, XMLElem parent);
+    XMLElem toXML(const ProductInfo& obj, XMLElem parent);
+    XMLElem toXML(const GeoInfo& obj, XMLElem parent);
+    XMLElem toXML(const MatchInformation& obj, XMLElem parent);
+
+    //! Read from XML object
+    void fromXML(const XMLElem collectionIDXML, CollectionID& collectionID);
+    void fromXML(const XMLElem globalXML, Global& global);
+    void fromXML(const XMLElem sceneCoordsXML, SceneCoordinates& scene);
+    void fromXML(const XMLElem dataXML, Data& data);
+    void fromXML(const XMLElem channelXML, Channel& channel);
+    void fromXML(const XMLElem pvpXML, Pvp& pvp);
+    void fromXML(const XMLElem DwellXML, Dwell& dwell);
+    void fromXML(const XMLElem refGeoXML, ReferenceGeometry& refGeo);
+    void fromXML(const XMLElem supportArrayXML, SupportArray& supportArray);
+    void fromXML(const XMLElem antennaXML, Antenna& antenna);
+    void fromXML(const XMLElem txRcvXML, TxRcv& txRcv);
+    void fromXML(const XMLElem errParamXML, ErrorParameters& errParam);
+    void fromXML(const XMLElem productInfoXML, ProductInfo& productInfo);
+    void fromXML(const XMLElem geoInfoXML, GeoInfo& geoInfo);
+    void fromXML(const XMLElem matchInfoXML, MatchInformation& matchInfo);
+
+
+    //! Create helper functions
+    void createParameterCollection(const std::string& name,
```

Comment:
Can you just use `SICommonXMLParser::addParameters()`?

Reply:
I found that after I defined this, and forgot to remove it.